apps/state/lib/state/facility.ex

```diff
@@ -89,11 +89,25 @@ defmodule State.Facility do
   defp do_searches([]), do: all()
 
   defp do_searches(search_operations) do
-    facilities =
-      Enum.flat_map(search_operations, fn search_operation ->
-        search_operation.()
+    search_results =
+      Stream.map(search_operations, fn search_operation ->
+        case search_operation.() do
+          results when is_list(results) ->
+            results
+
+          _ ->
+            []
+        end
       end)
 
-    Enum.uniq_by(facilities, & &1.id)
+    [first_result] = Enum.take(search_results, 1)
+
+    search_results
+    |> Stream.drop(1)
+    |> Enum.reduce(first_result, fn results, acc ->
+      acc_set = MapSet.new(acc)
+      Enum.filter(results, fn facility -> facility in acc_set end)
+    end)
+    |> Enum.uniq_by(& &1.id)
```

Comment:
Since you're already creating a MapSet, maybe we can be more efficient?
```elixir
Enum.reduce(search_results, :no_results, fn
  results, %MapSet{} = acc -> MapSet.intersection(acc, MapSet.new(results))
  results, :no_results -> MapSet.new(results)
end)
```

Reply:
This definitely looks better, but unfortunately causes a Dialyzer error:

```
    apps/state/lib/state/facility.ex:105: The call 'Elixir.MapSet':intersection(_acc@1::#{'__struct__':='Elixir.MapSet', _=>_},'Elixir.MapSet':t(_)) does not have an opaque term of type 'Elixir.MapSet':t(_) as 1st argument
```

This seems to be a known issue, but I don't know that there's any resolution on either the Elixir or the Dialyzer side: https://github.com/elixir-lang/elixir/issues/8463